specification/hdinsight/resource-manager/Microsoft.HDInsight/preview/2015-03-01-preview/applications.json

```diff
@@ -141,33 +123,21 @@
                         "$ref": "./examples/CreateApplication.json"
                     }
                 },
-                "parameters": [{
-                        "name": "resourceGroupName",
-                        "in": "path",
-                        "required": true,
-                        "type": "string",
-                        "description": "The name of the resource group."
+                "parameters": [
+                    {
+                        "$ref": "#/parameters/SubscriptionIdParameter"
                     },
                     {
-                        "name": "clusterName",
-                        "in": "path",
-                        "required": true,
-                        "type": "string",
-                        "description": "The name of the cluster."
+                        "$ref": "#/parameters/ResourceGroupNameParameter"
                     },
                     {
-                        "name": "applicationName",
-                        "in": "path",
-                        "required": true,
-                        "type": "string",
-                        "description": "The constant value for the application name.",
-                        "enum": [
-                            "hue"
-                        ],
-                        "x-ms-enum": {
-                            "name": "ApplicationName",
-                            "modelAsString": false
-                        }
+                        "$ref": "#/parameters/ClusterNameParameter"
+                    },
+                    {
+                        "$ref": "#/parameters/ApplicationNameParameter"
+                    },

```

Comment:
```
{
  "id": "1016",
  "code": "ConstantStatusHasChanged",
  "message": "The 'constant' status changed from the old version to the new.",
  "jsonref": "#/paths/~1subscriptions~1{subscriptionId}~1resourceGroups~1{resourceGroupName}~1providers~1Microsoft.HDInsight~1clusters~1{clusterName}~1applications~1{applicationName}/put/applicationName",
  "json-path": "#/paths/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.HDInsight/clusters/{clusterName}/applications/{applicationName}/put/applicationName",
  "type": "Error"
}
```

This is a breaking change in SDKs in each language.  The previous version had Enum with a single value "hue". The new one have a free form string. Relates to Put and Delete operations.

Reply:
This should not have been an enum in the first place. The service accepted values other than "hue" before the enum constraint was introduced, and this would break them. Additionally, the values for application name are dynamic, so it wasn't appropriate to use enums for them.